src/libcommonserver/io/iohelper_win.cpp

```diff
@@ -651,11 +651,11 @@ bool IoHelper::getRights(const SyncPath &path, bool &read, bool &write, bool &ex
     if (ioError != IoError::Success) {
         return isExpectedError(ioError);
     }
-    const bool isSymlink = itemType.linkType == LinkType::Symlink;
 
     std::error_code ec;
-    std::filesystem::perms perms =
-            isSymlink ? std::filesystem::symlink_status(path, ec).permissions() : std::filesystem::status(path, ec).permissions();
+    std::filesystem::perms perms = itemType.linkType == LinkType::Symlink || itemType.linkType == LinkType::Junction
```

Comment:
Since the computation of the permissions is not trivial, a small and well-named helper function would improve readability and allow us to test easily these specific conditionals.

Reply:
Done